core/src/main/java/org/elasticsearch/index/mapper/TextFieldMapper.java

```diff
@@ -124,7 +124,9 @@ public Builder fielddataFrequencyFilter(double minFreq, double maxFreq, int minS
         @Override
         public TextFieldMapper build(BuilderContext context) {
             if (positionIncrementGap != POSITION_INCREMENT_GAP_USE_ANALYZER) {
-                if (fieldType.indexOptions().compareTo(IndexOptions.DOCS_AND_FREQS_AND_POSITIONS) < 0) {
+                if (fieldType.indexOptions().compareTo(IndexOptions.DOCS_AND_FREQS_AND_POSITIONS) < 0 &&
+                        context.indexCreatedVersion() != null &&
+                        context.indexCreatedVersion().onOrAfter(Version.V_5_0_0_alpha1)) {
```

Comment:
Hrm, except we don't as of 5.0? Index created version can only be overriden in tests.